spade-aws/src/main/scala/com/salesforce/mce/spade/aws/action/SnsAlarm.scala

```diff
@@ -0,0 +1,43 @@
+package com.salesforce.mce.spade.aws.action
+
+import java.util.UUID
+
+import io.circe.syntax._
+
+import com.salesforce.mce.spade.aws.spec.SnsActionSpec
+import com.salesforce.mce.spade.workflow.Action
+
+case class SnsAlarm private (
+  topicArn: String,
+  id: String,
+  name: String,
+  subject: String,
+  message: String
+) {
+
+  def withSubject(s: String) = copy(subject = s)
+
+  def withMessage(s: String) = copy(message = s)
+
+  def asAction = Action(
+    id,
+    name,
+    "SnsAction",
```

Comment:
@realstraw is this intended action type?

Reply:
It's the class name after `com.salesforce.mce.orchard.io.`. In this case it would be `aws.action.SnsAction `.

Reply:
then why don't make it 
`val clz = Class.forName(s"com.salesforce.mce.orchard.io.aws.action.${conf.actionType}$$")`
 in ActionIO?

Reply:
the core service does not dictate on the provider. You can write non-aws provider too.

Reply:
ok. added the action type for sns.